charms/worker/k8s/src/charm.py

```diff
@@ -741,11 +745,19 @@ def _announce_kubernetes_version(self):
                 if not unit_version:
                     raise ReconcilerError(f"Waiting for version from {unit.name}")
                 if unit_version != local_version:
-                    # NOTE: Add a check to validate if we are doing an upgrade
-                    status.add(ops.WaitingStatus("Upgrading the cluster"))
-                    return
+                    waiting_units[relation_type] += 1
             relation.data[self.app]["version"] = local_version
 
+        if any(waiting_units.values()):
+            waiting_parts = []
+            if waiting_units["peer"]:
+                waiting_parts.append(f"{waiting_units['peer']} Control Plane")
+            if waiting_units["worker"]:
+                waiting_parts.append(f"{waiting_units['worker']} Worker")
```

Comment:
This is user-facing, right? Let's do the extra effort for pluralized words then:

```
if any(waiting_units.values()):
    # Build the status message for waiting units
    role_map = {
        "peer": "Control Plane",
        "worker": "Worker"
    }

    # Helper function to pluralize role names
    def pluralize(count, name):
        return f"{name}s" if count > 1 else name

    waiting_parts = [
        f"{count} {pluralize(count, role_map[role])}"
        for role, count in waiting_units.items() if count
    ]
    
    # Construct the status message
    status_msg = ", ".join(waiting_parts)
    
    # Update status and raise an error
    status.add(ops.WaitingStatus(f"Waiting for {status_msg} to upgrade"))
    raise ReconcilerError(f"Upgrade delayed: Waiting for {status_msg}")
 ```